### ISSUE TITLE
Adding COntinuos COin Betting (COCOB) Backprop optimizer

### ISSUE THREAD

**Issue by rehanguha** [tensorflow/addons#2064]
**Describe the feature and the current behaviour/state.**
Already a PR is raised against this issue: https://github.com/tensorflow/addons/pull/2063

**Relevant information**
- Are you willing to contribute it (yes/no): yes
- Are you willing to maintain it going forward? (yes/no): yes
- Is there a relevant academic paper? (if so, where): https://arxiv.org/pdf/1705.07795.pdf
- Is there already an implementation in another framework? (if so, where): https://github.com/harryliew/COCOB-optimizer/blob/master/cocob_bp.py (PyTorch Implementation) 
- Was it part of tf.contrib? (if so, where): No

**Which API type would this fall under (layer, metric, optimizer, etc.)**
optimizer

**Who will benefit with this feature?**
COCOB Algorithm's convergence is proven for convex and quasi-convex. But as I have experimented I saw Covex function performs great in comparison to other functions.

**Any other info.**
Found this code after my implementation and later adopted some parts from it as well. https://github.com/bremen79/cocob/blob/master/optimizer/cocob_optimizer.py

**Comment by bhack** [comment]
I think that we need to start to handle some of topics emerged in https://github.com/tensorflow/addons/issues/2045.

@tensorflow/sig-addons-maintainers

**Comment by rehanguha** [comment]
> I think that we need to start to handle some of topics emerged in #2045.
> 
> @tensorflow/sig-addons-maintainers

I have submitted the PR for this code already (65%-70% is complete) and it resides here https://github.com/tensorflow/addons/pull/2063
It will be less effort for me to push this first then pick something up from the list you provided. Meanwhile, I will go through them.

**Comment by bhack** [comment]
Never stop at title :wink:.
Start from https://github.com/tensorflow/addons/issues/2045#issuecomment-664373626

**Comment by rehanguha** [comment]
@bhack In the monthly SIG Addons meeting I brought this up and we all came to a conclusion that TensorFlow(core) optimizer documentation for sparse is not adequate. So I felt to push the PR #2063 and get community support for the sparse implementation. When it is done I will add it to the documentation for custom optimizers. I have rechecked the code and it is working as expected for dense implementation.
I personally feel we should go ahead and push this the way it is and not wait for completing the sparse implementation.
